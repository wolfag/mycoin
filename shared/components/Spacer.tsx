import React from 'react';
import { View } from 'react-native';

type Props = {
  size?: number;
  vertical?: boolean;
  full?: boolean;
};
const Spacer = ({ size = 12, vertical, full }: Props) => {
  return (
    <View
      style={{
        flex: full ? 1 : 0,
        height: vertical ? size : 0,
        width: vertical ? 0 : size,
      }}
    />
  );
};

export default Spacer;
