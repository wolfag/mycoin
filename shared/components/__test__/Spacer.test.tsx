import { render } from '@testing-library/react-native';

import Spacer from '../Spacer';

describe('Spacer', () => {
  it('renders default spacer', () => {
    const { toJSON } = render(<Spacer />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders vertical spacer', () => {
    const { getByTestId } = render(<Spacer vertical size={20} />);
    expect(getByTestId('Spacer')).toHaveStyle({
      height: 20,
      width: 0,
      flex: 0,
    });
  });

  it('renders full flex spacer', () => {
    const { getByTestId } = render(<Spacer full />);
    expect(getByTestId('Spacer')).toHaveStyle({
      height: 0,
      width: 12,
      flex: 1,
    });
  });
});
