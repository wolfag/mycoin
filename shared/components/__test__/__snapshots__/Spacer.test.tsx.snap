// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacer renders default spacer 1`] = `
<View
  style={
    {
      "flex": 0,
      "height": 0,
      "width": 12,
    }
  }
  testID="Spacer"
/>
`;
